test/integration/analytical_rosenbrock.cpp

```diff
@@ -13,120 +15,107 @@ using SparseMatrixTest = micm::SparseMatrix<double>;
 
 TEST(AnalyticalExamples, Troe)
 {
-  test_analytical_troe<micm::RosenbrockSolver<micm::Matrix, SparseMatrixTest>>();
+  auto builder = micm::CpuSolverBuilder<micm::RosenbrockSolverParameters>(micm::RosenbrockSolverParameters::ThreeStageRosenbrockParameters());
```

Comment:
```suggestion
  auto rosenbrock = micm::CpuSolverBuilder<micm::RosenbrockSolverParameters>(micm::RosenbrockSolverParameters::ThreeStageRosenbrockParameters());
  auto backward_euler = micm::CpuSolverBuilder<micm::BackwardEulerParameters>(micm::BackwardEulerParameters::BackwardEulerParameters());
  test_analytical_troe(rosenbrock);
  test_analytical_troe(backward_euler);
```

Is something like this possible now? If so, can we add it?

Reply:
It is now possible, but when I applied the analytical tests to the Backward Euler solver, none of them passed. They mostly ended up with answers that were close but not within the 1.0e-5 tolerance that we use in the tests.

I will open up a separate issue for testing the Backward Euler solver with the analytical systems.

Reply:
Note that the backward euler solver has a lower accuracy than the rosenbrock solver. Thus it is possible that it fails the same criteria used by the rosenbrock solver.